File: frontend/src/app/core/models/review.ts
export interface ReviewResponse {
  id: number;
  productId: number;
  userId: number;
  rating: number;
  title: string;
  comment: string;
  reviewerName: string;
  createdAt: string;
}

export interface ReviewRequest {
  userId: number;
  productId: number;
  rating: number;
  title: string;
  comment: string;
}

export interface ReviewSummary {
  productId: number;
  avgRating: number;
  reviewCount: number;
}

File: review-service/src/main/java/com/ecommerce/review_service/model/Review.java
package com.ecommerce.review_service.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "reviews", uniqueConstraints =
        @UniqueConstraint(columnNames = {"user_id", "product_id"}))
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "product_id", nullable = false)
    private Long productId;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(nullable = false)
    private Integer rating;

    @Column(length = 255)
    private String title;

    @Column(length = 2000)
    private String comment;

    @Column(name = "reviewer_name")
    private String reviewerName;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    public void prePersist() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    public void preUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
